inc/WC_JumiaPay_Purchase.php

```diff
@@ -83,58 +71,61 @@ private function getBasket() {
+            ],
+            "consumer" => [
+              "emailAddress" => sanitize_email($this->order->get_billing_email()),
+              "ipAddress" => sanitize_text_field($this->order->get_customer_ip_address()),
+              "country" => sanitize_text_field($this->countryCode),
+              "mobilePhoneNumber" => wc_sanitize_phone_number($this->order->get_billing_phone()),
+              "language" => $this->language,
+              "name" => sanitize_text_field(substr($this->order->get_billing_first_name()." ".$this->order->get_billing_last_name(), 0, 100)),
+              "firstName" => sanitize_text_field(substr($this->order->get_billing_last_name(), 0, 50)),
+              "lastName" => sanitize_text_field(substr($this->order->get_billing_first_name(), 0, 50))
+            ],
             "basket"=> [
-                "shipping"=>sanitize_text_field($this->order->get_shipping_tax()),
+                "shippingAmount"=>sanitize_text_field($this->order->get_shipping_tax()),
                 "currency"=>$this->currency,
-                "basketItems"=>$this->getBasket(),
-                "totalAmount"=> sanitize_text_field($this->order->get_total()),
-                "discount"=>sanitize_text_field($this->order->get_discount_total()),
-            ],
-            "consumerData"=> [
-                "emailAddress"=> sanitize_email($this->order->get_billing_email()),
-                "mobilePhoneNumber"=> wc_sanitize_phone_number($this->order->get_billing_phone()),
-                "country"=> sanitize_text_field($this->countryCode),
-                "firstName"=> sanitize_text_field($this->order->get_billing_first_name()),
-                "lastName"=> sanitize_text_field($this->order->get_billing_last_name()),
-                "ipAddress"=> sanitize_text_field($this->order->get_customer_ip_address()),
-                "dateOfBirth"=> "",
-                "language"=> $this->language,
-                "name"=> sanitize_text_field($this->order->get_billing_first_name()." ".$this->order->get_billing_last_name())
+                "items"=>$this->getBasket(),
             ],
-            "priceCurrency"=> $this->currency,
-            "priceAmount"=> sanitize_text_field($this->order->get_total()),
-            "purchaseReturnUrl"=>  esc_url_raw($this->baseUrl."/wc-api/payment_return/?orderid=".$this->order->get_id()),
-            "purchaseCallbackUrl"=> esc_url_raw($this->baseUrl."/wc-api/payment_callback/?orderid=".$this->order->get_id()),
             "shippingAddress"=> [
-                "addressLine1"=> sanitize_text_field($this->order->get_shipping_address_1()),
-                "addressLine2"=> sanitize_text_field($this->order->get_shipping_address_2()),
-                "city"=> sanitize_text_field($this->order->get_shipping_city()),
+                "addressPrimary"=> sanitize_text_field($this->order->get_shipping_address_1()),
+                "addressSecondary"=> sanitize_text_field($this->order->get_shipping_address_2()),
```

Comment:
Please validate up to 512 chars as in the documentation. Apply to billing Address too.

Reply:
done